tidy3d/plugins/smatrix/component_modelers/terminal.py

```diff
@@ -236,6 +236,12 @@ def port_VI(port_out: AbstractTerminalPort, sim_data: SimulationData):
         Z_numpy = port_impedances.transpose(*PortDataArray._dims).values.reshape(
             (len(self.freqs), len(port_names), 1)
         )
+
+        # Check for negative real part of port impedance and adjust V and Z accordingly
+        negative_real_Z = np.real(Z_numpy) < 0
+        V_numpy = np.where(negative_real_Z, -V_numpy, V_numpy)
+        Z_numpy = np.where(negative_real_Z, -Z_numpy, Z_numpy)
```

Comment:
Maybe add a sanity check to make sure that the sign choice is consistent over all frequencies. The impedance should not change sign as the frequency changes (at least for typical transmission lines)